frontend/src/metabase/visualizations/components/settings/ChartSettingFieldsPartition.jsx

```diff
@@ -44,148 +33,118 @@ class ChartSettingFieldsPartition extends React.Component {
       );
     }
   };
-  updateDisplayedValue = displayedValue =>
-    this.setState({
-      displayedValue: _.mapObject(displayedValue, cols =>
-        cols.map(col => col.field_ref),
-      ),
-    });
-  commitDisplayedValue = () => {
-    const { displayedValue } = this.state;
-    if (displayedValue != null) {
-      this.props.onChange(displayedValue);
-      this.setState({ displayedValue: null });
-    }
+
+  getPartitionType = partitionName => {
+    return partitionName === "rows" || partitionName === "columns"
+      ? "dimension"
+      : "metric";
```

Comment:
Do you think using switch...case would make this easier to understand?

```suggestion
    switch (partitionName) {
      case "rows":
      case "columns":
        return "dimension";
      default:
        return "metric";
    }
```

It doesn't make much different though given this code is still fairly short.